### PR TITLE
Turbine rework

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Turbines rework, better management, heat and pressure calculations, multiblock. Still a lot to do.
This turbine will be a 3x2 multi machine that don't uses open air to draw in gases but it has input and output ports, it will be added to the station with a cooling and heating loops and a burn chamber.
It will have a custom interface with informations, buttons and interactables numbers
The multi block works by having the controller in one corner, the main turbine body along one side and 2 mod blocks 
There will be different mod blocks with multiple systems that will help the machine in different ways, like more power generation at the cost of energy required, less energy required but less power, more efficient cooling, turbo chargers, etc. Those will be build not from frames, but from direct crafting (might change this)
image
yes the sprites will change (also the setup is from early testing so ignore it)
To do:
- [ ] Add machine building/deconstruction
- [ ] Add mods integration
- [ ] Modify maps
- [ ] Better code/docs/var names
- [ ] Better sprites
- [ ] TGUI work

## Why It's Good For The Game
Better turbines, will also start moving the project of SM rework forward.

## Changelog
:cl:
balance: Turbine rework (will write better CL later)
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
Ignore the commented out code, it will be removed in the next couple of commits

**Comment by tralezab** [comment]
twat will love those sprites bro

**Comment by Ratvar** [comment]
You got a big shaft there~

**Comment by Twaticus** [comment]
😶

**Comment by Berend-Broesder** [comment]
> yes the sprites will change

Don't, they're fantastic

**Comment by Ghilker** [comment]
I will get back to this, I promise

**Comment by Ghilker** [comment]
we back to it

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -279,9 +279,9 @@
 
 /obj/machinery/portable_atmospherics/canister/fusion_test/create_gas()
 	air_contents.add_gases(/datum/gas/hydrogen, /datum/gas/tritium)
-	air_contents.gases[/datum/gas/hydrogen][MOLES] = 300
-	air_contents.gases[/datum/gas/tritium][MOLES] = 300
-	air_contents.temperature = 10000
+	air_contents.gases[/datum/gas/hydrogen][MOLES] = 1000
+	air_contents.gases[/datum/gas/tritium][MOLES] = 1000
+	air_contents.temperature = 100000000
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/modules/atmospherics/machinery/portable/canister.dm:302]: It shouldn't but I find myself using it a lot when testing atmos components

[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/atmospherics/machinery/portable/canister.dm:302]: well then it should, since you're using it, right?

**Comment by Ghilker** [comment]
i swear i'll finish this

**Pull request closed without merging**